server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -132,8 +131,7 @@
             }
             if (custom.getRequiredFeature().isPresent()) {
                 final String requiredFeature = custom.getRequiredFeature().get();
-                // if it is a transport client we are lenient yet for a connected node it must have the required feature
-                return out.hasFeature(requiredFeature) || out.hasFeature(TransportClient.TRANSPORT_CLIENT_FEATURE) == false;
+                return out.hasFeature(requiredFeature);
```

Comment:
There are changes needed for the Javadoc here. It might be worth grepping the codebase for “transport client”.

Reply:
I did another pass and removed lots of uses. I rewrote some comments referring to the transport client as TODOs to consider modifying the code in question, but leave those for followups.